internal/command/deploy/deploy.go

```diff
@@ -263,8 +267,17 @@ func DeployWithConfig(ctx context.Context, appConfig *appconfig.Config, forceYes
 		}
 	}
 
+	httpFailover := flag.GetHTTPFailover(ctx)
+	usingWireguard := flag.GetWireguard(ctx)
+
 	// Fetch an image ref or build from source to get the final image reference to deploy
-	img, err := determineImage(ctx, appConfig)
+	img, err := determineImage(ctx, appConfig, usingWireguard)
+	if err != nil && usingWireguard && httpFailover {
+		span.SetAttributes(attribute.String("builder.failover_error", err.Error()))
+		span.AddEvent("using http failover")
+		img, err = determineImage(ctx, appConfig, false)
```

Comment:
this is really interesting. 

but, there are so many reasons why `determineImage` could fail, not just because of wireguard.

does it make sense to only fail over if a specific error happens(tunnel unavailble, hearbeat issues, etc) or can we just proceed as is? (i'm just wondering)

Reply:
hm we could go more in depth to only fail over on specific issues, but i think this is fine for two reasons:
1. determineImage is pretty atomic, I can't think of any "in-between" states an image could be in
2. because of that, even if the error isn't related to wireguard, retrying really can't hurt, to see if the error was just transient.

if you disagree though, I have no issue only failing over on specific errors :)

Reply:
no hard opinions actually, lets ship.